### PR TITLE
switch T3W1 support to U5

### PR DESCRIPTION
**Pull request by TychoVrahe**
This PR switches support T3W1 support from F4 (devkit 1) to U5 (prototype board rev A). 

Along with that:
- support for U5A is replaced with U5G
- support for DISC2 is switch to newer variant with U5G MCU
- flash layout for models on U5G is adjusted to fit better
- 32MHz HSE support is added for U5.
- U5 I2C driver is extended to support 4 instances of I2C

The board definition for revA is incomplete but should enable building on it.

### CODE REVIEW

File: core/embed/models/T3W1/boards/trezor_t3w1_revA.h [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+#define BACKLIGHT_PWM_TIM_CHANNEL TIM_CHANNEL_1
+#define BACKLIGHT_PWM_TIM_CCR CCR1
+#define BACKLIGHT_PWM_PIN GPIO_PIN_9
+#define BACKLIGHT_PWM_PORT GPIOB
+#define BACKLIGHT_PWM_PORT_CLK_EN __HAL_RCC_GPIOB_CLK_ENABLE
+
+#define NPM1300_I2C_INSTANCE 3
+
+#define I2C_COUNT 4
+#define I2C_INSTANCE_0 I2C3
```
Comment (cepetr) [inline core/embed/models/T3W1/boards/trezor_t3w1_revA.h:24]: How about sorting the I2C indexes like this:

- `I2C_INSTANCE_0` -> `I2C1`
- `I2C_INSTANCE_1` -> `I2C2`
- `I2C_INSTANCE_2` -> `I2C3`
- `I2C_INSTANCE_3` -> `I2C4`

File: core/embed/models/T3W1/boards/trezor_t3w1_revA.h [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+#define I2C_INSTANCE_3_SCL_CLK_EN __HAL_RCC_GPIOG_CLK_ENABLE
+#define I2C_INSTANCE_3_RESET_REG &RCC->APB1RSTR1
+#define I2C_INSTANCE_3_RESET_BIT RCC_APB1RSTR1_I2C1RST
+#define I2C_INSTANCE_3_EV_IRQHandler I2C1_EV_IRQHandler
+#define I2C_INSTANCE_3_ER_IRQHandler I2C1_ER_IRQHandler
+#define I2C_INSTANCE_3_EV_IRQn I2C1_EV_IRQn
+#define I2C_INSTANCE_3_ER_IRQn I2C1_ER_IRQn
+#define I2C_INSTANCE_3_GUARD_TIME 0
+
+#define TOUCH_SENSITIVITY 0x40
```
Comment (cepetr) [inline core/embed/models/T3W1/boards/trezor_t3w1_revA.h:96]: The touch control used on the T3W1 has no sensitivity register (like the FT36x6). It's a minor problem for now, but we should address it later.

**Review by cepetr — CHANGES_REQUESTED** [review]

**Comment by TychoVrahe** [comment]
@cepetr please re-review from https://github.com/trezor/trezor-firmware/pull/4344/commits/db91145433e24f5f3da80935a79db3bc38aa1fcc

File: core/embed/models/T3W1/boards/trezor_t3w1_revA.h [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+#define I2C_INSTANCE_3_SCL_PORT GPIOD
+#define I2C_INSTANCE_3_SCL_PIN GPIO_PIN_12
+#define I2C_INSTANCE_3_SCL_CLK_EN __HAL_RCC_GPIOD_CLK_ENABLE
+#define I2C_INSTANCE_3_RESET_REG &RCC->APB1RSTR2
+#define I2C_INSTANCE_3_RESET_BIT RCC_APB1RSTR2_I2C4RST
+#define I2C_INSTANCE_3_EV_IRQHandler I2C4_EV_IRQHandler
+#define I2C_INSTANCE_3_ER_IRQHandler I2C4_ER_IRQHandler
+#define I2C_INSTANCE_3_EV_IRQn I2C4_EV_IRQn
+#define I2C_INSTANCE_3_ER_IRQn I2C4_ER_IRQn
+#define I2C_INSTANCE_3_GUARD_TIME 0
```
Comment (cepetr) [inline core/embed/models/T3W1/boards/trezor_t3w1_revA.h:94]: Don't we need a 50 µs guard time here for optiga?

**Review by cepetr — APPROVED** [review]
I've re-reviewed it and everything seems fine. I just added a minor note to I2C3 about a potentially missing guard time.

File: core/embed/models/T3W1/boards/trezor_t3w1_revA.h (hunk 3 above)
Comment (TychoVrahe) [reply core/embed/models/T3W1/boards/trezor_t3w1_revA.h:94]: ultimately, yes. though neither in this file nor in the python definition we do have optiga set to be used yet, so lets handle it along with adding it

**Pull request merged**